src/test/java/com/epam/prejap/teatrees/block/TBlockTest.java

```diff
@@ -0,0 +1,87 @@
+package com.epam.prejap.teatrees.block;
+
+import com.epam.prejap.teatrees.TeaTrees;
+import com.epam.prejap.teatrees.game.Playfield;
+import com.epam.prejap.teatrees.game.Printer;
+import com.epam.prejap.teatrees.game.Waiter;
+import com.epam.prejap.teatrees.player.Player;
+import com.epam.prejap.teatrees.player.RandomPlayer;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+
+import static org.testng.Assert.assertEquals;
+
+@Test
+public class TBlockTest {
+
+    private TBlock tBlock;
+    private ByteArrayOutputStream outputStream;
+    TeaTrees teatrees;
+    BlockFeed blockFeed;
+    Printer printer;
+    Playfield playfield;
+    Waiter waiter;
+    Player player;
+
+    @BeforeMethod
+    private void setup() {
+        tBlock = new TBlock();
+        outputStream = new ByteArrayOutputStream();
+        System.setOut(new PrintStream(outputStream));
+        blockFeed = new BlockFeed();
+        printer = new Printer(System.out);
+        playfield = new Playfield(10, 20, blockFeed, printer);
+        player = new RandomPlayer();
+        waiter = new Waiter(10);
+        teatrees = new TeaTrees(playfield, waiter, player);
+    }
+
+    /**
+     * Method that intercepts console output and checks if it matches the desired output (block shapes).
+     * It plays the game using the play() method and executes the game in a default mode when you don't pass
+     * a parameter to the play method, or uses a specified block when you use a parametrized method
+     * (e.g. 0 if using a default square block, 1 if using T-block, etc.)
+     */
+    public void shouldHave3HashesInSecondRowAnd1InThirdRow(){
+        //given
+        var game = teatrees.play(1);
```

Comment:
Instead of passing a block index to the play method (which makes the signature harder to understand), it would be much better to have an interface for BlockFeed functionality and be able to mock it in the tests to return just the desired block.

Reply:
done